sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -0,0 +1,182 @@
+    private let credential: CommunicationUserCredential
+    private let options: AzureCommunicationChatClientOptions
+    private let service: AzureCommunicationChatService
+
+    // MARK: Initializers
+
+    /// Create a ChatClient.
+    /// - Parameters:
+    ///   - endpoint: The Communication Services endpoint.
+    ///   - credential: The user credential.
+    ///   - options: Options used to configure the client.
+    public init(
+        endpoint: String,
+        credential: CommunicationUserCredential,
+        withOptions options: AzureCommunicationChatClientOptions
+    ) throws {
+        self.endpoint = endpoint
+        self.credential = credential
+        self.options = options
+
+        guard let endpointUrl = URL(string: endpoint) else {
+            throw AzureError.client("Unable to form base URL.")
+        }
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        let client = try AzureCommunicationChatClient(
+            endpoint: endpointUrl,
+            authPolicy: authPolicy,
+            withOptions: options
+        )
+
+        self.service = client.azureCommunicationChatService
+    }
+
+    // MARK: Public Methods
+
+    /// Create a ChatThreadClient for the ChatThread with id threadId.
+    /// - Parameters:
+    ///   - forChatThread: The threadId.
+    public func createClient(forChatThread threadId: String) throws -> ChatThreadClient {
+        return try ChatThreadClient(
+            endpoint: endpoint,
+            credential: credential,
+            threadId: threadId,
+            withOptions: options
+        )
+    }
+
+    /// Create a new ChatThread and return the ChatThreadClient for it.
+    /// - Parameters:
+    ///   - chatThread: Request for creating a chat thread with the topic and members to add.
+    ///   - withOptions: Create chat thread options.
+    ///   - completionHandler: A completion handler that receives a ChatThreadClient on success.
+    public func create(
+        chatThread: CreateChatThreadRequest,
+        withOptions options: AzureCommunicationChatService.CreateChatThreadOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<ChatThreadClient>
+    ) {
+        service.create(chatThread: chatThread, withOptions: options) { result, httpResponse in
+            do {
+                switch result {
+                case let .success(chatThread):
+                    guard let threadId = chatThread.id else {
+                        throw AzureError.service("Service response does not contain ThreadId.")
+                    }
+
+                    // Create and return a ChatThreadClient for the new ChatThread
+                    let chatThreadClient = try ChatThreadClient(
+                        endpoint: self.endpoint,
+                        credential: self.credential,
+                        threadId: threadId,
+                        withOptions: self.options
+                    )
```

Comment:
I still disagree with this. By the principle of least surprise, I would not expect creating an object to return a client for that object. How does someone create a chat thread without creating a client?

Reply:
I agree its not intuitive to return the client - I found out that there is a proposal to decouple the two but this is currently the implementation across the other SDKs so I was being consistent with that until its decided to make the change across the board.

Reply:
Other Azure SDKs tend to have something like `myServiceClient.getItemClient()` or similar. Especially since you have a `get` method that returns a `ChatThread`, this should either also return a `ChatThread` or return nothing.

Reply:
Most commonly I have seen create methods return the object they created.

Reply:
Updated to return ChatThread instead of the client. I believe there is upcoming work for the SDKs to refactor this anyways, we can deal with any new changes then